streams/src/main/java/org/apache/kafka/streams/processor/internals/SourceNode.java

```diff
@@ -112,7 +106,7 @@ public String toString() {
     /**
      * @return a string representation of this node starting with the given indent, useful for debugging.
      */
-    public String toString(final String indent) {
+    public String toString(final String indent, final List<String> topics) {
```

Comment:
We do not need to pass in the topics here, just make the topic string creation logic inside `ProcessorTopology` since the data is stored there.